modules/miner/update.go

```diff
@@ -8,24 +8,19 @@ import (
 )
 
 // ProcessConsensusDigest will update the miner's most recent block.
-func (m *Miner) ProcessConsensusDigest(revertedIDs, appliedIDs []types.BlockID) {
+func (m *Miner) ProcessConsensusChange(cc modules.ConsensusChange) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
-	// Sanity check - the length of appliedIDs should always be non-zero.
-	if build.DEBUG && len(appliedIDs) == 0 {
-		panic("received a digest with no applied blocks")
-	}
-
 	// Adjust the height of the miner.
-	m.height -= types.BlockHeight(len(revertedIDs))
-	m.height += types.BlockHeight(len(appliedIDs))
+	m.persist.Height -= types.BlockHeight(len(cc.RevertedBlocks))
+	m.persist.Height += types.BlockHeight(len(cc.AppliedBlocks))
 
 	// Update the unsolved block.
 	var exists1, exists2 bool
-	m.unsolvedBlock.ParentID = appliedIDs[len(appliedIDs)-1]
-	m.target, exists1 = m.cs.ChildTarget(m.unsolvedBlock.ParentID)
-	m.unsolvedBlock.Timestamp, exists2 = m.cs.MinimumValidChildTimestamp(m.unsolvedBlock.ParentID)
+	m.persist.UnsolvedBlock.ParentID = cc.AppliedBlocks[len(cc.AppliedBlocks)-1].ID()
+	m.persist.Target, exists1 = m.cs.ChildTarget(m.persist.UnsolvedBlock.ParentID)
+	m.persist.UnsolvedBlock.Timestamp, exists2 = m.cs.MinimumValidChildTimestamp(m.persist.UnsolvedBlock.ParentID)
```

Comment:
it's a little weird to have all these fields prefixed with `persist` now. Oh well